runtime/include/atomics/standard/chpl-atomics.h

```diff
@@ -0,0 +1,300 @@
+////                START OF REAL ATOMICS EXCHANGE OPS                    ////
+//////////////////////////////////////////////////////////////////////////////
+#define DECLARE_REAL_ATOMICS_EXCHANGE_OPS(type) \
+static inline type atomic_exchange_explicit_ ## type(atomic_ ## type * obj, type _value, memory_order order) { \
+  return atomic_exchange_explicit(obj, _value, order); \
+} \
+static inline type atomic_exchange_ ## type(atomic_ ## type * obj, type value) { \
+  return atomic_exchange(obj, value); \
+} \
+static inline chpl_bool atomic_compare_exchange_strong_explicit_ ## type(atomic_ ## type * obj, type expected, type desired, memory_order order) { \
+  return atomic_compare_exchange_strong_explicit(obj, &expected, desired, order, order); \
+} \
+static inline chpl_bool atomic_compare_exchange_strong_ ## type(atomic_ ## type * obj, type expected, type desired) { \
+  return atomic_compare_exchange_strong(obj, &expected, desired); \
+} \
+static inline chpl_bool atomic_compare_exchange_weak_explicit_ ## type(atomic_ ## type * obj, type expected, type desired, memory_order order) { \
+  return atomic_compare_exchange_weak_explicit(obj, &expected, desired, order, order); \
+} \
+static inline chpl_bool atomic_compare_exchange_weak_ ## type(atomic_ ## type * obj, type expected, type desired) { \
+  return atomic_compare_exchange_weak(obj, &expected, desired); \
+}
+
+
+///////////////////////////////////////////////////////////////////////////////
+////                   START OF REAL ATOMICS FETCH OPS                    ////
+//////////////////////////////////////////////////////////////////////////////
+#define DECLARE_REAL_ATOMICS_FETCH_OPS(type) \
+static inline type atomic_fetch_add_explicit_ ## type(atomic_ ## type * obj, type operand, memory_order order) { \
+  type old_val = *obj; \
+  type new_val; \
+  do { \
+    new_val = old_val + operand; \
+  } while (!atomic_compare_exchange_weak_explicit(obj, &old_val, new_val, order, order)); \
+  return old_val; \
+} \
+static inline type atomic_fetch_add_ ## type(atomic_ ## type * obj, type operand) { \
+  return atomic_fetch_add_explicit_ ## type(obj, operand, memory_order_seq_cst); \
+} \
+static inline type atomic_fetch_sub_explicit_ ## type(atomic_ ## type * obj, type operand, memory_order order) { \
+  type old_val = *obj; \
+  type new_val; \
+  do { \
+    new_val = old_val - operand; \
+  } while (!atomic_compare_exchange_weak_explicit(obj, &old_val, new_val, order, order)); \
+  return old_val; \
+} \
+static inline type atomic_fetch_sub_ ## type(atomic_ ## type * obj, type operand) { \
+  return atomic_fetch_sub_explicit_ ## type(obj, operand, memory_order_seq_cst); \
+}
+
+
+// Actually declare the atomics for integer and real types using the above macros
+DECLARE_ATOMICS_BASE(bool, chpl_bool);
+DECLARE_ATOMICS_EXCHANGE_OPS(bool, chpl_bool);
+
+
+#define DECLARE_ATOMICS(type) \
+  DECLARE_ATOMICS_BASE(type, type) \
```

Comment:
The real fetch ops had to be different, but I was able to consolidate the base and exchange macros.  It passes my initial tests.  I'm going to pound harder on it to make sure.  I'll add another comment when that testing is complete.